Description: Upstream changes introduced in version 1.0-1ubuntu4
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 t10-keyboard (1.0-1ubuntu4) natty; urgency=low
 .
   * fixed a problem with old profile format
 .
 The person named in the Author field signed this changelog entry.
Author: Nicolai Ommer <nicolai.ommer@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- t10-keyboard-1.0.orig/t10-keyboard
+++ t10-keyboard-1.0/t10-keyboard
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-java -jar /usr/share/t10-keyboard/t10-keyboard.jar
+java -jar /usr/share/t10-keyboard/t10-keyboard.jar $@
--- t10-keyboard-1.0.orig/src/edu/dhbw/t10/manager/profile/ProfileManager.java
+++ t10-keyboard-1.0/src/edu/dhbw/t10/manager/profile/ProfileManager.java
@@ -87,7 +87,7 @@ public class ProfileManager {
 		loadConfig();
 
 		// fill activeProfileName and profilePathes with the data from the config object
-		loadSerializedProfiles(); // deserializes all profiles, fills profiles
+		loadProfiles(); // deserializes all profiles, fills profiles
 
 		// if no profiles were loaded, create a new one
 		if (profiles.size() == 0) {
@@ -368,7 +368,7 @@ public class ProfileManager {
 	 * 
 	 * @author SebastianN
 	 */
-	private void loadSerializedProfiles() {
+	private void loadProfiles() {
 		int counter = 0;
 		if (profiles == null) {
 			profiles = new ArrayList<Profile>();
@@ -389,14 +389,19 @@ public class ProfileManager {
 		//deserializing the profiles
 		for (File profileFile : profileFiles) {
 			Properties prop = new Properties();
+			Profile prof;
 			try {
 				FileInputStream fis = new FileInputStream(profileFile);
 				prop.load(fis);
+				prof = new Profile(prop);
+				profiles.add(prof);
 			} catch (IOException err) {
-				logger.warn("Could not deserialize profile file");
+				// prof = new Profile("toDelete, take the new profile format", datapath);
+				logger.warn("Could not read the profile file");
+			} catch (ExceptionInInitializerError e) {
+				// TODO DirkK or Anybody else: handle old profiles
 			}
-			Profile prof = new Profile(prop);
-			profiles.add(prof);
+
 		}
 		logger.info("Deserialized " + counter + " profiles.");
 	}
--- t10-keyboard-1.0.orig/src/edu/dhbw/t10/type/profile/Profile.java
+++ t10-keyboard-1.0/src/edu/dhbw/t10/type/profile/Profile.java
@@ -42,7 +42,7 @@ import edu.dhbw.t10.type.tree.PriorityTr
  */
 public class Profile implements Serializable {
 	/**  */
-	private static final long			serialVersionUID	= 5085464540715301877L;
+	private static final long			serialVersionUID	= 5085464540715301878L;
 	// --------------------------------------------------------------------------
 	// --- variables and constants ----------------------------------------------
 	// --------------------------------------------------------------------------
@@ -106,6 +106,10 @@ public class Profile implements Serializ
 	
 	public Profile(Properties prop) {
 		properties = prop;
+		if (!properties.containsKey("name")) {
+			logger.error("Tried to load profile with invalid properties");
+			throw new ExceptionInInitializerError();
+		}
 		load();
 	}
 
--- t10-keyboard-1.0.orig/src/res/version
+++ t10-keyboard-1.0/src/res/version
@@ -1 +1 @@
-v1.0-11-g6a334ff
+v1.0-14-gffe169b
